scripts/ros-generate-cache.sh

```diff
@@ -0,0 +1,132 @@
+# /bin/sh (deliberately no !#)
+#
+# Usage: cd meta-ros
+#        sh scripts/ros-generate-cache.sh ROS_DISTRO YYYYMMDD PATH-TO-LOCAL-ROSDISTRO ROSDISTRO-COMMIT
+#            or
+#        sh scripts/ros-generate-cache.sh --version
+#
+#        Generate cache.yaml file for the YYYYMMDD release of ROS_DISTRO from the specified commit of a local
+#        ros/rosdistro.git . If running prior to first release of ROS_DISTRO, specify "none" for YYYYMMDD. NB. The release
+#        date might not match the commit timestamp of ROSDISTRO-COMMIT; however, the commit timestamp is used for the
+#        DATETIME when forming the name of the created branch.
+#
+#        It will also copy the files from PATH-TO-LOCAL-ROSDISTRO/rosdep to the meta-ros and asks you to
+#        adjust /etc/ros/rosdep/sources.list.d/20-default.list before running rosdep update if it doesn't
+#        point to these files already.
+#
+# Copyright (c) 2019 LG Electronics, Inc.
+
+SCRIPT_NAME="ros-generate-cache"
+SCRIPT_VERSION="1.0.0"
+
+usage() {
+    echo "Usage: cd meta-ros"
+    echo "Usage: sh scripts/$SCRIPT_NAME.sh ROS_DISTRO YYYYMMDD PATH-TO-LOCAL-ROSDISTRO ROSDISTRO-COMMIT"
+    echo "               or"
+    echo "       sh scripts/$SCRIPT_NAME.sh --version"
+    exit 1
+}
+
+if [ $1 = "--version" ]; then
+    echo "$SCRIPT_NAME $SCRIPT_VERSION"
+    exit
+fi
+
+[ $# -ne 4 ] && usage
+
+ROS_DISTRO=$1
+ROS_DISTRO_RELEASE_DATE=$2
+ROS_DISTRO_CHECKOUT=$3
+ROS_DISTRO_COMMIT=$4
+
+case $ROS_DISTRO in
+    "kinetic"|"melodic")
+        export ROS_VERSION="1"
+        export ROS_PYTHON_VERSION="2"
+        ;;
+
+    "crystal"|"dashing")
+        export ROS_VERSION="2"
+        export ROS_PYTHON_VERSION="3"
+        ;;
+
+    *)  echo "ABORT: Unrecognized ROS_DISTRO: $ROS_DISTRO"
+        exit 1
+        ;;
+esac
+
+case $ROS_DISTRO_RELEASE_DATE in
+    none|[2-9][0-9][0-9][0-9][0-1][0-9][0-3][0-9])
+        : OK
+        ;;
+
+    *)  echo "ABORT: ROS_DISTRO_RELEASE_DATE not YYYYMMDD or 'none': '$3'"
+        exit 1
+        ;;
+esac
+
+if [ ! -d $ROS_DISTRO_CHECKOUT ]; then
+    echo "ABORT: '$ROS_DISTRO_CHECKOUT' not found"
+    exit 1
+fi
+
+if [ -n "$(git status --porcelain=v1)" ]; then
+    echo "ABORT: Uncommitted modifications detected by Git -- perhaps invoke 'git reset --hard'?"
+    exit 1
+fi
+
+cd $ROS_DISTRO_CHECKOUT
+path_to_rosdistro=$PWD
+ROS_DISTRO_COMMIT=$(git rev-list -1 $ROS_DISTRO_COMMIT) || { echo "ABORT: commit '$ROS_DISTRO_COMMIT' doesn't exist in '$ROS_DISTRO_CHECKOUT'"; exit 1; }
+
+if [ -n "$(git status --porcelain=v1)" ]; then
```

Comment:
There's no need to force the local ros/rosdistro directory to be "clean", as we won't be creating any commits for it.

Reply:
ok